src/test/java/com/github/romankh3/image/comparison/BaseTest.java

```diff
@@ -26,4 +26,13 @@ public void assertImagesEqual(BufferedImage expected, BufferedImage actual) {
         }
     }
 
+    protected Throwable getException(Runnable action) {
```

Comment:
That's super weird stuff. If you actually want to verify the exception — catch it and check the content. It's not `Spock` and such ugly hacks will not it make like `Spock`.

There 2 ways I'd recommend:
1. migrate all tests to Spock and be happy.
2. migrate to jUnit5 and write tests in a way the framework proposes. I'd also recommend trying the `Truth` library.

Reply:
It's a good reason to use JUnit5 here.